app/main/presenters/search_results.py

```diff
@@ -4,17 +4,26 @@
 class SearchResults(object):
     """Provides access to the search results information"""
 
-    def __init__(self, response):
+    def __init__(self, response, lots_by_slug):
         self.search_results = response['services']
-        self._add_highlighting()
+        self._lots = lots_by_slug
+        self._annotate()
         self.total = response['meta']['total']
         if 'page' in response['meta']['query']:
             self.page = response['meta']['query']['page']
 
-    def _add_highlighting(self):
-        for index, service in enumerate(self.search_results):
-            if 'highlight' in service:
-                if 'serviceSummary' in service['highlight']:
-                    self.search_results[index]['serviceSummary'] = Markup(
-                        ''.join(service['highlight']['serviceSummary'])
-                    )
+    def _annotate(self):
+        for service in self.search_results:
+            self._replace_lot(service)
+            self._add_highlighting(service)
+
+    def _replace_lot(self, service):
+        # replace lot slug with reference to dict containing all the relevant lot data
+        service['lot'] = self._lots.get(service['lot'])
```

Comment:
I think this is probably neater inline if there's only one reference to it, which it looks like there is.